kuwala/core/canvas/src/pages/DataSourcePreview.js

```diff
@@ -1,151 +1,364 @@
-                                <th className={'bg-kuwala-light-green text-white lowercase px-4 py-2'}>Header 3</th>
-                                <th className={'bg-kuwala-light-green text-white lowercase px-4 py-2'}>Header 4</th>
-                                <th className={'bg-kuwala-light-green text-white lowercase px-4 py-2'}>Header 5</th>
-                            </tr>
-                            </thead>
-                            <tbody>
-                                {[...Array(100).keys()].map((e) => {
-                                    console.log(e)
-                                    return (
-                                        <tr>
-                                            <td className={'text-left px-4 py-2 border border-kuwala-light-green'}>THINGS</td>
-                                            <td className={'text-left px-4 py-2 border border-kuwala-light-green'}>THINGS</td>
-                                            <td className={'text-left px-4 py-2 border border-kuwala-light-green'}>THINGS</td>
-                                            <td className={'text-left px-4 py-2 border border-kuwala-light-green'}>THINGS</td>
-                                            <td className={'text-left px-4 py-2 border border-kuwala-light-green'}>THINGS</td>
-                                        </tr>
-                                    )
-                                })}
-                            </tbody>
-                        </table>
+                    <div className={'flex flex-col bg-white w-9/12 overflow-auto'}>
+                        {selectedTable
+                            ?
+                                isTableDataPreviewLoading
+                                    ?
+                                        <div className="flex flex-col w-full h-full justify-center items-center">
+                                            <div
+                                                className="spinner-border animate-spin inline-block w-24 h-24 border-4 text-kuwala-green rounded-full"
+                                                role="status">
+                                                <span className="visually-hidden">Loading...</span>
+                                            </div>
+                                        </div>
+                                    :
+                                        renderTableDataPreview(tableDataPreview.columns, tableDataPreview.rows)
+                            :
+                            <div className="flex flex-col w-full h-full text-xl font-light justify-center items-center">
+                                <p>Select a table from the <span className={'text-kuwala-green'}>left</span></p>
+                                <p>to preview the data</p>
+                            </div>
+                        }
```

Comment:
Putting this in a separate render function would make the code more readable IMO.